src/assets/scripts/client/scope/ScopeModel.js

```diff
@@ -0,0 +1,208 @@
+import _has from 'lodash/has';
+import _isNil from 'lodash/isNil';
+import EventBus from '../lib/EventBus';
+import NavigationLibrary from '../navigationLibrary/NavigationLibrary';
+import RadarTargetCollection from './RadarTargetCollection';
+import { EVENT } from '../constants/eventNames';
+import { DECIMAL_RADIX } from '../constants/globalConstants';
+import { THEME } from '../constants/themes';
+
+export default class ScopeModel {
+    constructor() {
+        this._eventBus = EventBus;
+        this._navigationLibrary = NavigationLibrary;
+        // TODO: Use this!
+        this._sectorCollection = [];
+        this._theme = THEME.DEFAULT;
+
+        this.radarTargetCollection = [];
+
+        this._init();
+    }
+
+    /**
+     * Complete initialization tasks
+     *
+     * @for ScopeModel
+     * @method _init
+     */
+    _init() {
+        this.radarTargetCollection = new RadarTargetCollection(this._theme);
+    }
+
+    /**
+     * Accept a pending handoff from another sector
+     *
+     * @for ScopeModel
+     * @method acceptHandoff
+     * @param radarTargetModel {RadarTargetModel}
+     * @return result {array} [success of operation, system's response]
+     */
+    acceptHandoff(radarTargetModel) {
+        return [false, 'acceptHandoff command not yet available'];
+    }
+
+    /**
+     * Amend the cruise altitude OR interim altitude for a given `RadarTargetModel`
+     *
+     * @for ScopeModel
+     * @method amendAltitude
+     * @param radarTargetModel {RadarTargetModel}
+     * @param altitude {string}
+     * @return result {array} [success of operation, system's response]
+     */
+    amendAltitude(radarTargetModel, altitude) {
+        altitude = parseInt(altitude, DECIMAL_RADIX);
+
+        return radarTargetModel.amendAltitude(altitude);
+    }
+
+    /**
+     * Enable handlers
+     *
+     * @for ScopeModel
+     * @method disable
+     */
+    disable() {
+        this._eventBus.off(EVENT.SET_THEME, this._setTheme);
+    }
+
+    /**
+     * Disable handlers
+     *
+     * @for ScopeModel
+     * @method enable
+     */
+    enable() {
+        this._eventBus.on(EVENT.SET_THEME, this._setTheme);
+    }
+
+    /**
+     * Initiate a handoff to another sector
+     *
+     * @for ScopeModel
+     * @method initiateHandoff
+     * @param radarTargetModel {RadarTargetModel}
+     * @param sectorCode {string} the handoff code for the receiving sector
+     * @return result {array} [success of operation, system's response]
+     */
+    initiateHandoff(radarTargetModel, sectorCode) {
+        return [false, 'initiateHandoff command not yet available'];
+    }
+
+    /**
+     * Change direction and/or length of data block leader line
+     *
+     * @for ScopeModel
+     * @method moveDataBlock
+     * @param radarTargetModel {RadarTargetModel}
+     * @param commandArguments {array}
```

Comment:
should this be: `array<string>`?

Reply:
hm. actually no, it's just a string. I reconfigured some stuff and didn't update the docblocks. Done now. 👍